Лабораторная работа 1. Доработка

TemZZZ opened the pull request.

Commit:
refactor(solution Lab1): переименовано из solution Lab1Model

fixture(project Lab1View): создан проект WindowsForms Lab1View

- добавлена форма MainForm
- добавлена форма AddRadioComponentForm

fixture(project RegularExoressionTextBoxLib): добавлен проект библиотеки элемента управления WindowsForms RegularExoressionTextBoxLib

- добавлен класс RegularExpressionTextBox, наследует TextBox

Commit:
fixture(RegularExpressionTextBox): добавлено автосвойство регулярного выражения

Commit:
fixture(RegularExpressionTextBox.RegularExpressionPattern): изменено автосвойство

- RegularExpressionPattern переименовано из RegularExpression
- тип автосвойства изменен с Regex на string

Commit:
fixture(Lab1View): добавлен проект Lab1View

- добавлена MainForm
- на MainForm добавлены addRadioComponentButton, deleteRadioComponentButton, radioComponentsDataGridView

fixture(RegularExpressionTextBox): реализованы конструкторы и обработчики событий

- реализованы конструкторы с параметрам string шаблона и с параметром объект Regex
- реализован конструктор без параметра
- реализован обработчик OnTextChanged для проверки корректности вводимых данных
- реализован обработчик OnLostFocus для проверки корректности введенных данных

Commit:
style(ConsoleLoader): константы R, L и C перенесены в метод GetComponent

style(ConsoleLoader): константы "+" и "-" перенесены в метод PrintComplex

Commit:
style(ConsoleLoader.GetComponent): изменен комментарий

Commit:
style(Main): удалены пустые строки между using

Commit:
style(Main): перегруппированы строки

- перегруппированы строки для лучшей прослеживаемости логики
- константа Q занесена в метод Main
- отдельный блок кода в начале цикла исключен

Commit:
style(ConsoleLoader.GetComponent): удалены пустые строки в блоке switch-case

style(ConsoleLoader.PrintComplex): блок if переписан в одну строку

iKalentii commented on ConsoleLoader/ConsoleLoader.cs:
```diff
@@ -104,20 +86,23 @@ public static class ConsoleLoader
         public static ComponentBase GetComponent(
             string type, Action<string> printer = null)
         {
+            /// <summary>Символ резистора</summary>
```
XML комменты из методов надо убрать + они не сильно нужны, т.к. из названия константы понятно - что это такое.

iKalentii commented on ConsoleLoader/ConsoleLoader.cs:
```diff
@@ -154,13 +139,15 @@ public static class ConsoleLoader
         /// представления</param>
         public static void PrintComplex(Complex value, Action<string> printer)
         {
+            /// <summary>Знак плюс</summary>
```
Убрать

iKalentii commented on ConsoleLoader/ConsoleLoader.cs:
```diff
@@ -154,13 +139,15 @@ public static class ConsoleLoader
-            {
-                sign = signMinus;
-            }
+            if (im < 0) { sign = signMinus; }
```
Зря перенёс на одну строку. Если больше одного слова - лучше делить на несколько строк.

iKalentii commented on ConsoleLoader/Main.cs:
```diff
@@ -1,62 +1,46 @@
-
 	public static void Main()
 	{
+		/// <summary>Символ выхода из программы</summary>
```
Убрать

iKalentii commented on ConsoleLoader/Main.cs:
```diff
@@ -1,62 +1,46 @@
+				userAnswer,	Console.WriteLine);
 			if (component == null) { continue; }
 
 			ConsoleLoader.AskComponentValue(in component, Console.Write);
```
Тут не совсем корректно выполнена работа с пользовательским вводом. 
![image](https://user-images.githubusercontent.com/902806/81639922-7719f480-9447-11ea-9ada-f03b745f399c.png)
Если вылетела ошибка программа не должна возвращаться в первое меню, она должна позвонить пользователю ввести параметр заново

Commit:
style(ConsoleLoader): удалены комментарии из методов, в блоках if добавлены фигурные скобки

Commit:
style(ConsoleLoader): удалена пустая строка между using

Commit:
style(Main): удален очевидный комментарий

Commit:
style(Main): код оформлен по RSDN

Commit:
feature(ConsoleLoader): добавлен метод DotAndCommaToNumberDecimalSeparator

Commit:
feature(StringToDouble): метод StringToDouble переписан с учетом DotAndCommaToNumberDecimalSeparator

Commit:
fix(ConsoleLoader): исправлен метод PrintComplex

- неверно выводилась бесконечность методом PrintComplex

Commit:
refactor(ConsoleLoader.GetRadioComponent): переименован из ConsoleLoader.GetComponent

Commit:
refactor(ConsoleLoader.AskRadioComponentValue): переименован из ConsoleLoader.AskComponentValue

Commit:
style(Main): длина строк кода теперь не более 78 символов

Commit:
feature(Main): переписана логика программы

- запрос типа радиокомпонента осуществляется в GetRadioComponentLoop
- запрос значения физической величины осуществляется в GetRadioComponentValueLoop
- запрос частоты осуществляется GetFrequencyLoop

iKalentii commented on ConsoleLoader/ConsoleLoader.cs:
```diff
@@ -13,20 +13,19 @@ namespace ConsoleLoaderModel
+        /// <returns></returns>
+        public static string DotAndCommaToNumberDecimalSeparator(string text)
+        {
+            string sep = CultureInfo.
```
Лучше конечно не сокращать. Времени сэкономил немного, а читаемость переменной убил)

iKalentii commented on ConsoleLoader/Main.cs:
```diff
@@ -1,64 +1,130 @@
+	/// <see cref="exitCharacter"/> в любом регистре
+	/// </summary>
+	/// <returns>Положительное <see cref="double"/> или
+	/// <see cref="double.IsNaN(double)"/></returns>
```
Стандартные типы данных нет смысла перечислять через теги <see cref>

Commit:
style(ConsoleLoader.DotAndCommaToNumberDecimalSeparator): sep переименован в regionalNumberDecimalSeparator

Commit:
style(ConsoleLoader.DotAndCommaToNumberDecimalSeparator): изменен комментарий

Commit:
style(ConsoleLoader.StringToDouble): в комментарии стандартные типы данных записаны без тэгов

Commit:
style(ConsoleLoader.IsPositive): в комментарии стандартные типы данных записаны без тэгов

Commit:
style(ConsileLoader.cs): дтрективы using пространства имен System записаны в алфавитном порядке

Commit:
style(MainApp.GetRadioComponentLoop): null в комментарии записан без тэгов

Commit:
style(MainApp.GetRadioComponentValueLoop): стандартные типы данных в комментарии записаны без тэгов

Commit:
style(MainApp.GetFrequencyLoop): стандартные типы данных в комментарии записаны без тэгов
